src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -109,5 +110,16 @@ private static Interop.libproc.rusage_info_v3 GetCurrentProcessRUsage()
         {
             return Interop.libproc.proc_pid_rusage(Environment.ProcessId);
         }
+
+        private static unsafe Interop.libSystem.mach_timebase_info_data_t GetTimeBase()
+        {
+            Interop.libSystem.mach_timebase_info_data_t timeBase = default;
+            var returnCode = Interop.libSystem.mach_timebase_info(&timeBase);
+            if (returnCode != 0)
+            {
+                throw new Win32Exception(returnCode);
```

Comment:
Please check if this is correct.

Reply:
One possibility I see here would be to just return a coefficient of `1` in case of an error in this place, to fall back to the old behavior.

Reply:
> One possibility I see here would be to just return an effective coefficient of numer: 1, denom: 1 in case of an error in this place, to fall back to the old behavior.

Yes, let's do this.

And, let's add a `Debug.Assert` so debug test runs will fail if this would ever return non-success.

Should we account for `mach_timebase_info` not existing on some systems?

Reply:
I think our minimum macOS support offsets this API’s minimum requirement.

Reply:
Apparently [it was available at least in OS X 10.10.1](https://stackoverflow.com/a/31454136/2684760) back in 2015, and .NET only supports 10.15+, so we should be okay.

I'll apply the suggestion.

Reply:
Done!